core/src/main/resources/error/delta-error-classes.json

```diff
@@ -614,6 +614,12 @@
     ],
     "sqlState" : "42000"
   },
+  "DELTA_FEATURES_PROTOCOL_METADATA_MISMATCH" : {
+    "message" : [
+      "Unable to operate on this table because the following table features are enabled in metadata but not listed in protocol: <features>."
```

Comment:
Thanks for the review!
This error message is not when a feature is not supported by the client, but when the client recognizes a table property belongs to a specific feature that is not listed in the protocol. Foe example:

- Protocol: `(3, 7, readFeatures=[], writerFeatures=[])`
- Metadata: `delta.changeDataFeed = true`

In this case the writer knows a feature `changeDataFeed` should be in the Protocol but isn't, hence this error message. This would not happen when a table is properly updated every time, but we can't be sure about 3rd party writers.

Reply:
Speaking of a solution -- currently there is no way to fix this table because any write attempt will hit this exception.
Ideally we should provide a Delta Config to bypass this consistency check and give the user a chance to fix the broken table. Of course, the whole history would be invalid in this case so users must be careful.